src/mode-toggle.ts

```diff
@@ -0,0 +1,136 @@
+import { getElement } from './utils';
+import * as styles from './styles';
+import { DisplayMode } from './indexer';
+
+/**
+ * The {@link DisplayModeToggle} class represents a UI component for switching between two
+ * display modes, structures and atoms, by clicking on the respective buttons.
+ *
+ * It automatically adjusts the height of the map element to ensure there is space for the toggle.
+ * The component uses Shadow DOM to encapsulate styles.
+ * A callback function (`onchange`) is triggered whenever the toggle value (atom / structure) is changed
+ */
+export class DisplayModeToggle {
+    /// Shadow root for isolation
+    private _shadow: ShadowRoot;
+    /// Toggle buttons container element
+    private _toggleContainer: HTMLElement;
+    /// Callback fired when the user changes the toggle value
+    public onchange: (mode: DisplayMode) => void;
+
+    /**
+     * Create a new {@link DisplayModeToggle} instance
+     *
+     * @param element HTML element or HTML id of the DOM element where the toggle will be attached
+     * @param isPerAtom flag indicating if the atom mode should be checked
+     */
+    constructor(element: string | HTMLElement, isPerAtom: boolean = true) {
```

Comment:
instead of `isPerAtom`, why not pass the `mode` around here? (same in the function below)